File: GFG/Modified Numbers and Queries/Modified Numbers and Queries.cpp
class Solution {
  public:
    int sumOfAll(int l, int r){
        // code here
        
        vector<int> p(10005, 0);
        p[0] = p[1] = 1;
        for(int i = 2; i <= 10004; i++){
            if(p[i] == 0){
             for(int j = i; j <= 10004; j += i){
                 p[j] += i;
             }
            }
        }
        int ans = 0;
        for(int i = l; i <= r; i++){
            ans += p[i];   
        }
        return ans;
    }
};

File: GFG/Modified Numbers and Queries/Modified Numbers And Queries.java
class Solution
{
    public int sumOfAll(int l, int r)
    {
        // code here
        int f[]=new int[r+1];
        int ans=0;
        if(l<=1) ans+=1;
        for(int i=2;i<=r;i++) {
            if(f[i]==0) {
                for(int j=i;j<=r;j+=i) {
                    f[j]+=i;
                }
            }
            if(i>=l) {
                if(f[i]==0) ans+=i;
                else ans+=f[i];
            }
        }
        return ans;
    }
}
